### PR TITLE
Fix Phpspec for symfony 6

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | partially #400 
| License         | MIT

### CODE REVIEW

File: src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php [hunk 1]
```diff
@@ -61,8 +63,10 @@ function it_sets_code_as_disabled_when_resource_is_not_new(
 
         $resource->getCode()->willReturn('Code12');
 
+        $form->add('code', Argument::any(), Argument::withEntry('disabled', true))->willReturn($form);
+
         $form
-            ->add('code', Argument::type('string'), Argument::withEntry('disabled', true))
+            ->add('code', Argument::any(), Argument::withEntry('disabled', true))
```
Comment (lchrusciel) [inline src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php]: Shouldn't it be?
```suggestion
            ->add('code', TextType::class, Argument::withEntry('disabled', true))
```

Comment (loic425) [reply src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php]: you're right, done.

File: src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php [hunk 2]
```diff
@@ -61,8 +63,10 @@ function it_sets_code_as_disabled_when_resource_is_not_new(
 
         $resource->getCode()->willReturn('Code12');
 
+        $form->add('code', TextType::class, Argument::withEntry('disabled', true))->willReturn($form);
```
Comment (lchrusciel) [inline src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php]:
```suggestion
```

File: src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php [hunk 3]
```diff
@@ -61,8 +63,10 @@ function it_sets_code_as_disabled_when_resource_is_not_new(
...
+
         $form
-            ->add('code', Argument::type('string'), Argument::withEntry('disabled', true))
+            ->add('code', TextType::class, Argument::withEntry('disabled', true))
```
Comment (lchrusciel) [inline src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php:66]:
```suggestion
            ->add('code', TextType::class, Argument::withEntry('disabled', true))
            ->willReturn($form)
```

File: src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php [hunk 4]
```diff
@@ -43,8 +43,10 @@ function it_sets_code_as_enabled_when_resource_is_new(FormEvent $event, FormInte
 
         $resource->getCode()->willReturn(null);
 
+        $form->add('code', Argument::any(), Argument::withEntry('disabled', false))->willReturn($form);
+
         $form
-            ->add('code', Argument::type('string'), Argument::withEntry('disabled', false))
+            ->add('code', Argument::any(), Argument::withEntry('disabled', false))
```
Comment (lchrusciel) [inline src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php]:
```suggestion
            ->add('code', TextType::class, Argument::withEntry('disabled', false))
            ->willReturn($form)
```

File: src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php [hunk 5]
```diff
@@ -43,8 +43,10 @@ function it_sets_code_as_enabled_when_resource_is_new(FormEvent $event, FormInte
 
         $resource->getCode()->willReturn(null);
 
+        $form->add('code', Argument::any(), Argument::withEntry('disabled', false))->willReturn($form);
```
Comment (lchrusciel) [inline src/Bundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php]:
```suggestion
```

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Loïc! :tada:

**Pull request merged**